### PR TITLE
Part 4 React Router

### PR DESCRIPTION
**Pull request by antonskrebetz**

### CODE REVIEW

File: src/components/app/app.jsx [hunk 1]
```diff
@@ -1,22 +1,33 @@
...
-      <SearchPage />
-      <ActorPage />
-      <MoviePage />
+      <Router>
+        <Header/>
+        <Container maxWidth="xl">
+          <Routes>
+            <Route path="/" element={[<ToggleButtons key={'toggle'}/>, <PopularMovies key={12345}/>]} />
+            <Route path="/toprated" element={[<ToggleButtons key={'toggle'}/>, <TopRatedMovies key={45465}/>]} />
+            <Route path="/upcoming" element={[<ToggleButtons key={'toggle'}/>,<UpcomingMovies key={12376}/>]} />
```
Comment (shubich) [inline src/components/app/app.jsx]: it does't look good

File: src/components/pagination/pagination.jsx [hunk 2]
```diff
@@ -1,22 +1,32 @@
...
+      navigate(`${url.pathname}?page=${page}`)
+    }
     window.scroll(0, 0);
-  }, [setPage]);
+  }, [navigate, query, url.pathname]);
 
   return (
     <ThemeProvider theme={DarkTheme}>
       <Pagination 
+        page={+actualPage || DEFAULT_PAGE}
```
Comment (shubich) [inline src/components/pagination/pagination.jsx:24]: why do you need `+` here?

File: src/services/use-query.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+import {useLocation} from 'react-router-dom';
+
+function useQuery() {
```
Comment (shubich) [inline src/services/use-query.js]: react-router-dom doesn't provide useQuery out of the box?

File: src/components/app/app.jsx (hunk 1 above)
Comment (shubich) [reply src/components/app/app.jsx]: I don't see any sense to place togglebuttons here, no profit in this, better place it inside other components

Comment (antonskrebetz) [reply src/components/app/app.jsx]: Fixed

File: src/components/pagination/pagination.jsx (hunk 2 above)
Comment (antonskrebetz) [reply src/components/pagination/pagination.jsx:24]: We get a string from the url address, the pagination component should take a number.

File: src/services/use-query.js (hunk 3 above)
Comment (antonskrebetz) [reply src/services/use-query.js]: Fixed

**Pull request merged**